resources/rulefiles/samples/simple.yml

```diff
@@ -0,0 +1,31 @@
+# For generic configurations that do not pertain to rules or scopes, but rather
+# on how LeGit analyzes and applies this file
+Config: {}
+
+# Rules that should be applied to all cases, regardless of scope
+GenericRules: {}
+
+# Rules with specific scopes attached to them
+ScopedRules:
+  # At least one scope required. This defines in which contexts to evaluate the
+  # rules
+  - Scope:
+      # In this example attached rules should apply to branches that have a
+      # number followed by an hiphen and arbitrary text (numerical or
+      # otherwise). The action tells LeGit to include this scope, so rules will
+      # be evaluated for branches which obey this scope
+      BranchNameScope:
+        # Field specific to BranchNameScope. Format string where "[]" are used
+        # as formatting delimiters
+        NameLike: '[number]-[text]'
+        # Wether to include when scope condition is met, or exclude. All scope
+        # conditions must have this Action specified
+        Action: include
+    # List of rules to evaluate, should the scope above be met
+    Rules:
```

Comment:
Doesn't this also need a " - " like the Scope?

Reply:
Yup, you're right. Furthermore, I'm noticing this now, and I don't know if it's legal YAML for each element in the list to have a different structure (I have, in `Rules`, an element with fields `Type` and `MaxSize`, but other rules won't have a `MaxSize` field). It's probably best to wrap each Rule/Scope in a simple structure of `Type` and `Fields`, where the `Fields` may change arbitrarily, or even be empty

Reply:
Updated as mentioned above. Used name "Parameters" instead of "Fields"

Data classes were unchanged under the assumption that the parser can correctly assign these values without requiring an actual array of "Parameters" to be stored in-code